src/lib/network_pool/transaction_pool.ml

```diff
@@ -926,260 +959,290 @@ struct
-                              in
-                              f_log
-                                "rejecting $cmd because we don't accept fees \
-                                 in $token"
-                                ~metadata:
-                                  [ ("cmd", User_command.to_yojson tx)
-                                  ; ("token", Token_id.to_yojson fee_token) ] ;
-                              go txs'' pool
-                                ( accepted
-                                , ( tx
-                                  , Diff_versioned.Diff_error
-                                    .Unwanted_fee_token )
-                                  :: rejected )
-                          | Error Invalid_transaction ->
-                              let%bind _ =
-                                trust_record
-                                  ( Trust_system.Actions.Sent_useless_gossip
-                                  , Some
-                                      ( "rejecting command because had \
-                                         invalid signature or was malformed"
-                                      , [] ) )
-                              in
-                              go txs'' pool
-                                ( accepted
-                                , ( tx
-                                  , Diff_versioned.Diff_error.Invalid_signature
-                                  )
-                                  :: rejected )
-                          | Error err ->
-                              let diff_err, error_extra =
-                                of_indexed_pool_error err
-                              in
-                              if is_sender_local then
-                                [%str_log' error t.logger]
-                                  (Rejecting_command_for_reason
-                                     { command= tx
-                                     ; reason= diff_err
-                                     ; error_extra }) ;
-                              let%bind _ =
-                                trust_record
-                                  ( Trust_system.Actions.Sent_useless_gossip
-                                  , Some
-                                      ( "rejecting $cmd because of $reason. \
-                                         ($error_extra)"
-                                      , [ ("cmd", User_command.to_yojson tx)
-                                        ; ("reason", yojson_fail_reason err)
-                                        ; ("error_extra", `Assoc error_extra)
-                                        ] ) )
-                              in
-                              go txs'' pool
-                                (accepted, (tx, diff_err) :: rejected) )
-                        else
+                    let new_ledger_hash = Base_ledger.merkle_root ledger in
+                    if not (Ledger_hash.equal ledger_hash new_ledger_hash) then
```

Comment:
When is the ledger mask updated? every time the ledger changes, best_tip_ledger gets a new mask. `ledger` here would only fail with dangling parent, no? Did you mean to get the updated ledger from `best_tip_ledger` again?

Reply:
This PR is in response to #7613, where the hypothesis is that the ledger is being mutated. This is to protect against that mutation.